client/src/components/RequestsView.js

```diff
@@ -0,0 +1,309 @@
+                    onClick={this.handleMenuClick}
+                >
+                    Confirmed Meetings
+                    {   (this.state.confirmed !== []) &&
+                        <Label color='teal'>{this.state.confirmed.length}</Label>
+                    }
+                </Menu.Item>
+                <Menu.Item
+                    id='completed'
+                    name='Completed Meetings'
+                    active={this.state.activeItem === 'completed'}
+                    onClick={this.handleMenuClick}
+                />
+            </Menu>
+            {this.state.activeItem === 'unconfirmed' &&
+                <div style={{paddingLeft: 13, paddingRight: 13}}>
+                    <RequestCards 
+                        confirmedTimes={this.state.confirmedTimes}
+                        activeSet={this.state.activeItem}
+                        requests={this.state.unconfirmed}
+                        liftRequests={this.handleStatusUpdate}
+                        timeOffset={this.state.timeOffset}
+                        userId={this.props.userDetails._id}
+                    />
+                </div>
+            }
+            {this.state.activeItem === 'confirmed' &&
+                <div style={{paddingLeft: 13, paddingRight: 13}}>
+                    <RequestCards 
+                        activeSet={this.state.activeItem}
+                        requests={this.state.confirmed}
+                        liftRequests={this.handleStatusUpdate}
+                        timeOffset={this.state.timeOffset}
+                        userId={this.props.userDetails._id}
+                    />
+                </div>
+            }
+            {this.state.activeItem === 'completed' &&
+                <div style={{paddingLeft: 13, paddingRight: 13}}>
+                    <RequestCards 
+                        activeSet={this.state.activeItem}
+                        requests={this.state.completed}
+                    />
+                </div>
+            }
+            </div>
+        )
+    }
+}
+
+
+/*
+ * DETAILS
+ * Child component - takes array of requests, constructs display
+ *                   shows valid actions for a given set
+ * PROPS
+ * Always:
+ * activeSet (string) - Shows what set is currently being used
+ * requests - array of relevant requests
+ * 
+ * Sometimes:
+ * userId - mongo ID for profile (used to update records)
+ * liftRequests (method) - lifts results of an update to parent component
+ * timeOffset
+ * confirmedTimes - array of meeting times that have already been confirmed
+ *                  used to disable approval, preventing double booking
+ */
+class RequestCards extends Component {
+    state={
+        requests: [],
+        display: []
+    }
+    // This allows the component to update its state should a prop value change
+    async componentWillReceiveProps({requests}) {
+        await this.setState({requests: requests})
```

Comment:
Is there a reason we're using `await` here (and in componentWillMount's setState)?

Reply:
In my experience, code relying on a state would not work as intended if it was set earlier in a synchronous function - that said, it looks like RequestCards only uses requests as a prop so it should be safe to remove the state and the set state, as well as the async

Reply:
Changed - in testing, componentWillMount can be synchronous and has been updated to reflect that
However, componentWillReceiveProps does need to be asynchronous and set state with an await. I have changed it to construct the display using the state to reflect this